### PR TITLE
refactor: update contents for coding conventions

### PR DESCRIPTION
**Pull request by Saransh-cpp**
- get rid of the weird `context` import and `conventions.py`
- showcase `ruff` and `pre-commit` (with `pre-commit.ci`)

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
The workflow changes in #240 should make everything pass

**Review by dpshelio — APPROVED** [review]
All good and beautiful!

File: ch05construction/conventions.py [hunk 1]
```diff
@@ -1,81 +0,0 @@
-### "dense"
```
Comment (dpshelio) [inline ch05construction/conventions.py:1]: Is this file only being used in `02conventions`?

**Review by dpshelio — CHANGES_REQUESTED** [review]
To check whether the conventions file is only used in this file.

File: ch05construction/conventions.py (hunk 1 above)
Comment (Saransh-cpp) [reply ch05construction/conventions.py:1]: Can confirm this is not being used anywhere. In fact, it is not being used in `02conventions` too. The variables in `02conventions` are coming from `context.py` -

https://github.com/UCL/rsd-engineeringcourse/blob/6741f65bbfdfbe09a755dafbbeff5ae879f17e7f/ch05construction/01introduction.ipynb.py#L133-L188

Comment (Saransh-cpp) [reply ch05construction/conventions.py:1]: Looking at this again, maybe I should not get rid of `from context import *`, given that it is present in a few other file as well and removing it from everywhere will result in a lot of repetition. Maybe I should just import specific things from the file and assume that the readers are going through the course serially so they know what `context.py` is.

**Review by dpshelio — APPROVED** [review]
:raised_hands:

**Pull request merged**